src/lib/gateway/gateway.ts

```diff
@@ -91,8 +91,9 @@ class Gateway extends GatewayBase {
     type: GatewayRequestType,
     request: GatewayRequest
   ) {
+    const txId = request.txId ?? (await this.getFirstUnusedTxId());
+    delete request.txId;
```

Comment:
why do you delete the `txid` from the request?

Reply:
Because it's not part of the native request sent to the server (it's sent through the `tx_id` property).